### PR TITLE
Mapping

### PR DESCRIPTION
**Pull request by sokrahta**
What's new
---------------
+ add get/put mapping to readme 
+ add `ignoreConflicts` boolean parameter, default `false`

[Elasticsearch reference: put mapping merge conflicts](http://www.elasticsearch.org/guide/en/elasticsearch/reference/current/indices-put-mapping.html#merging-conflicts)

[Review on Reviewable](https://reviewable.io/reviews/gphat/wabisabi/21)

### CODE REVIEW

File: src/main/scala/wabisabi/Client.scala [hunk 1]
```diff
@@ -300,8 +300,10 @@ class Client(esURL: String) extends Logging {
    * @param type The type name to which the mappings will be applied.
    * @param body The mapping.
    */
-  def putMapping(indices: Seq[String], `type`: String, body: String): Future[Response] = {
-    val req = (url(esURL) / indices.mkString(",") / `type` / "_mapping").setBody(body.getBytes(StandardCharsets.UTF_8))
+  def putMapping(indices: Seq[String], `type`: String, body: String, ignoreConflicts: Boolean = false): Future[Response] = {
```
Comment (gphat) [inline src/main/scala/wabisabi/Client.scala]: Please document the new `ignoreConflicts` param in the Scaladoc above!

File: .gitignore [hunk 2]
```diff
@@ -1,5 +1,5 @@
+*~
 project/project
-project/target
```
Comment (gphat) [inline .gitignore]: Why these changes?

File: src/main/scala/wabisabi/Client.scala [hunk 3]
```diff
@@ -300,8 +300,10 @@ class Client(esURL: String) extends Logging {
...
+    val req = (url(esURL) / indices.mkString(",") / `type` / "_mapping").
+      setQueryParameters(Map("ignore_conflicts" -> List(ignoreConflicts.toString))).
+      setBody(body.getBytes(StandardCharsets.UTF_8))
```
Comment (gphat) [inline src/main/scala/wabisabi/Client.scala]: I hunted around and can't find any definitive information on what's considered style-appropriate for Scala when it comes to line-wrapping chained methods. I, however, prefer leading dots. Could you change that here?

**Comment by gphat** [comment]
Thanks for this! I've got a few nits here but would be very pleased to merge and release this once you've fixed those up.

File: .gitignore (hunk 2 above)
Comment (sokrahta) [inline .gitignore]: not so important, but:
`*~` ignores text editor temp files such as `Client.scala~`
`project/target` is a sub case of `target/`

**Comment by sokrahta** [comment]
Hey @gphat thanks for the comments, I made some changes in response.
Cheers

File: .gitignore (hunk 2 above)
Comment (gphat) [inline .gitignore]: Gotya!

**Comment by gphat** [comment]
Thanks @sokrahta !

**Comment by gphat** [comment]
Released as 2.0.17!

**Comment by sokrahta** [comment]
groovy!
